src/stacks-hierarchy/Root_PurchaseHarborSearchScreen/Root_PurchaseHarborSearchScreen.tsx

```diff
@@ -0,0 +1,216 @@
+    route.params;
+
+  const {t} = useTranslation();
+  const inputRef = useRef<InternalTextInput>(null);
+  const [text, setText] = useState('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const onSave = (selectedStopPlace: StopPlace) => {
+    selectedStopPlace &&
+      navigation.navigate({
+        name: 'Root_PurchaseOverviewScreen',
+        params: {
+          mode: 'Ticket',
+          fareProductTypeConfig,
+          preassignedFareProduct,
+          fromHarbor: fromHarbor ?? selectedStopPlace,
+          toHarbor: fromHarbor ? selectedStopPlace : undefined,
+        },
+        merge: true,
+      });
+  };
+
+  const styles = useStyles();
+  // capturing focus on mount and on press
+  const focusInput = () => setTimeout(() => inputRef.current?.focus(), 0);
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    if (isFocused) focusInput();
+  }, [isFocused]);
+
+  const {harbors, isLoading, error} = useGetHarbors(fromHarbor?.id);
+
+  useEffect(() => {
+    if (error) {
+      setErrorMessage(translateErrorType(error, t));
+    }
+  }, [error]);
+
+  const debouncedText = useDebounce(text, 200);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.headerContainer}>
+        <FullScreenHeader
+          title={
+            fromHarbor
+              ? t(HarborSearchTexts.header.titleTo)
+              : t(HarborSearchTexts.header.titleFrom)
+          }
+          leftButton={{type: 'back'}}
+        />
+      </View>
+      <View style={styles.header}>
+        <View style={styles.withMargin}>
+          <TextInputSectionItem
+            ref={inputRef}
+            radius="top-bottom"
+            label={
+              fromHarbor
+                ? t(HarborSearchTexts.stopPlaces.to)
+                : t(HarborSearchTexts.stopPlaces.from)
+            }
+            value={text}
+            onChangeText={setText}
+            showClear={Boolean(text?.length)}
+            onClear={() => setText('')}
+            placeholder={t(HarborSearchTexts.searchField.placeholder)}
+            autoCorrect={false}
+            autoComplete="off"
+            testID="searchInput"
+          />
+        </View>
+      </View>
+      <ScrollView
+        style={styles.scroll}
+        contentContainerStyle={styles.contentBlock}
+        keyboardShouldPersistTaps="handled"
+        onScrollBeginDrag={() => Keyboard.dismiss()}
+      >
+        {isLoading && <ActivityIndicator />}
+        {error && (
+          <View style={styles.errorMessage}>
+            <ScreenReaderAnnouncement message={errorMessage} />
+            <MessageBox type="warning" message={errorMessage} />
+          </View>
+        )}
+
+        <HarborResults
+          harbors={harbors}
+          onSelect={onSave}
+          searchText={debouncedText}
+          fromHarborName={fromHarbor?.name}
+        />
+      </ScrollView>
+    </View>
+  );
+};
+
+function translateErrorType(
```

Comment:
I saw this same function is on many other places duplicated, would be great to refactor this :) but is up to you this time, would be a good improvement.

Reply:
Pushed suggestion for refactor now